### PR TITLE
Exit voice heartbeat loop eventually after closing VWS

### LINKED ISSUE #93: Voice heartbeat fiber not stopped after closing connection
Disconnecting from voice by sending a status update and closing the socket:
```cr
client.voice_state_update(guild, nil, false, false)
voice_client.close
```
results in the following traceback:
```
Unhandled exception in spawn:
Closed socket (IO::Error)
0x4e6e6b: *CallStack::unwind:Array(Pointer(Void)) at ??
0x647d30: check_open at /usr/lib/crystal/http/web_socket.cr 61:5
0x647cd8: send at /usr/lib/crystal/http/web_socket.cr 64:5
0x646de5: send at /home/snapcase/devel/crystal/wat/lib/discordcr/src/discordcr/websocket.cr 35:5
0x6a2c72: heartbeat_loop at /home/snapcase/devel/crystal/wat/lib/discordcr/src/discordcr/voice.cr 102:11
0x4e2f49: ??? at /usr/lib/crystal/kernel.cr 0:3
0x503e2e: run at /usr/lib/crystal/fiber.cr 255:3
0x4d49f6: ~proc2Proc(Fiber, (IO::FileDescriptor | Nil)) at /usr/lib/crystal/concurrent.cr 0:3
0x0: ??? at ??
```
which seems to be [Discord::VoiceClient#heartbeat_loop](https://github.com/meew0/discordcr/blob/master/src/discordcr/voice.cr#L99-L108) trying to keep the connection alive after disconnecting.

### PR DESCRIPTION
**Pull request by z64**
Expands the `WebSocket#close` delegation to set a heartbeat-sending flag, similar to what we do in `Client`.

Fixes #93

### CODE REVIEW

File: src/discordcr/voice.cr [hunk 1]
```diff
@@ -104,6 +109,7 @@ module Discord
     private def heartbeat_loop
       loop do
         if @heartbeat_interval
+          break unless @send_heartbeats
```
Comment (RX14) [inline src/discordcr/voice.cr]: `while @send_heartbeats` instead of `loop do`?

File: src/discordcr/voice.cr [hunk 2]
```diff
@@ -35,6 +35,7 @@ module Discord
     @token : String
 
     @heartbeat_interval : Int32?
+    @send_heartbeats : Bool?
```
Comment (RX14) [inline src/discordcr/voice.cr]: `@send_heartbeats = false`?

**Comment by meew0** [comment]
Can you squash this?

**Comment by z64** [comment]
Done

**Pull request merged**